File: ClientApp/src/app/shared/models/getEventsParameter.ts
import { GetEventsSort } from './getEventsSort';

export interface IGetEventsParameter {
    latitude?: number;
    longitude?: number;
    from?: Date;
    to?: Date;
    str?: string;
    userId?: string;
    tags?: number[];
    page?: number;
    count?: number;
    sort?: GetEventsSort;
}

export class GetEventsParameter implements IGetEventsParameter {
  constructor(
    public latitude: number = 0,
    public longitude: number  = 0,
    public from: Date = null,
    public to: Date = null,
    public str: string = '',
    public userId: string = '',
    public tags: number[] = [],
    public page: number = 0,
    public count: number = 20,
    public sort: GetEventsSort = GetEventsSort.latest
  ) { }
}

File: Models/TransferModels/GetEventsParameter.cs
using System;
using System.Collections.Generic;

namespace ServerApi.Models.TransferModels
{
    public class GetEventsParameter
    {
        //public IPoint Location { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public DateTime? From { get; set; }
        public DateTime? To { get; set; }
        public List<int> Tags { get; set; }
        public string UserId { get; set; }
        public string Str { get; set; }
        public int Page { get; set; }
        public int Count { get; set; }
        public GetEventsSort Sort { get; set; }
    }

    public enum GetEventsSort
    {
        Latest,
        Popular,
        Nearest
    }
}